qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/QuteAssert.java

```diff
@@ -787,14 +789,56 @@ public static void assertLocation(List<? extends Location> actual, Location... e
 		assertArrayEquals(expected, actual.toArray());
 	}
 
+	// ------------------- Rename assert
+
+	public static void assertRename(String value, String newText) throws BadLocationException {
+		assertRename(value, newText, FILE_URI, PROJECT_URI, TEMPLATE_BASE_DIR, Collections.emptyList());
+	}
+
+	public static void assertRename(String value, String newText, List<TextEdit> expectedEdits)
+		throws BadLocationException {
+		assertRename(value, newText, FILE_URI, PROJECT_URI, TEMPLATE_BASE_DIR, expectedEdits);
+	}
+
+	public static void assertRename(String value, String newText, String fileUri, String projectUri,
+		String templateBaseDir, List<TextEdit> expectedEdits) throws BadLocationException {
+		int offset = value.indexOf('|');
+		value = value.substring(0, offset) + value.substring(offset + 1);
+
+		QuteProjectRegistry projectRegistry = new MockQuteProjectRegistry();
+		Template template = createTemplate(value, fileUri, projectUri, templateBaseDir, projectRegistry);
+		QuteLanguageService languageService = new QuteLanguageService(new JavaDataModelCache(projectRegistry));
+
+		Position position = template.positionAt(offset);
+
+		WorkspaceEdit workspaceEdit = languageService.doRename(template, position, newText, () -> {
+		});
+		try {
+			List<TextEdit> actualEdits = workspaceEdit.getChanges().get(FILE_URI);
+			assertArrayEquals(expectedEdits.toArray(), actualEdits.toArray());
+		} catch (NullPointerException e) {
+			assertArrayEquals(expectedEdits.toArray(), Collections.emptyList().toArray());
+		}
+	}
+
+	public static List<TextEdit> edits(String newText, Range... ranges) {
+		List<TextEdit> re = new ArrayList<TextEdit>();
+		for (Range range : ranges) {
+			re.add(te(range.getStart().getLine(), range.getStart().getCharacter(), range.getEnd().getLine(),
+				range.getEnd().getCharacter(), newText));
+			// re.add(new TextEdit(range, newText))

```

Comment:
Is the `te` helper preferred over creating a `TextEdit` instance, or vice versa?

Reply:
I suggest that you writ ethe method like this:

```
private static List<TextEdit> edits(String newText, Range... ranges) {
		return Stream.of(ranges).map(r -> new TextEdit(r, newText)).collect(Collectors.toList());
	}
```